Remove Notary and Content Trust

dave-tucker opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/linuxkit/linuxkit/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
This commit removes Notary and Content Trust.
Notary v1 is due to be replaced with Notary v2 soon.
There is no clean migration path from one to the other.
For now, this removes all signing from LinuxKit.
We will look to add this back once a new Notary alternative
becomes available.

**- How I did it**
- Removed any command line flags relating to signing and content trust
- Removed `trust.go`
- Removed all references to signing and content trust from makefiles, scripts and documentation
- Ensured that everything still compiles

**- How to verify it**
If CI is still green, then all should be well.

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->
- Removed support for Docker Content Trust and Notary Signing

**- A picture of a cute animal (not mandatory but encouraged)**
![image](https://user-images.githubusercontent.com/3781777/111803360-e281ed80-88c6-11eb-9487-f503a4decb65.png)

deitch commented:
Hats off @dave-tucker  I will look early next week.

rn commented:
Thanks. I'll take a look later as well. May I suggest that to merge this PR it requires approval from all active maintainers which currently seem to be @deitch @justincormack and myself to make sure we all (publicly) agree with this (@dave-tucker I assume your approval is implicit in making this PR in the first place)

dave-tucker commented:
Yes @rn, my +1 is implicit

deitch commented on kernel/Makefile:
```diff
@@ -144,7 +138,7 @@ forcepush: forcepush_$(2)$(3)$(4)
 
 # 'docker build' with the FROM image supplied as --build-arg
-# *and* with DOCKER_CONTENT_TRUST=1 currently does not work
+# *and* with  currently does not work
```
This line is odd.

deitch commented on src/cmd/linuxkit/moby/images.go:
```diff
@@ -39,16 +37,5 @@ func imagePull(ref *reference.Spec, alwaysPull bool, trust bool, cacheDir string
 // imageLayoutWrite takes an image name and pulls it down, writing it locally
 func imageLayoutWrite(cacheDir string, ref *reference.Spec, architecture string, trust bool) (ImageSource, error) {
```
We removed how this uses `trust`, but kept it as an arg. I think we should just get rid of it here, too.

deitch commented on src/cmd/linuxkit/build.go:
```diff
@@ -223,7 +217,7 @@ func build(args []string) {
 
 		log.Infof("Create outputs:")
-		err = moby.Formats(filepath.Join(*buildDir, name), image, buildFormats, size, !*buildDisableTrust, cacheDir)
+		err = moby.Formats(filepath.Join(*buildDir, name), image, buildFormats, size, false, cacheDir)
```
Why do we pass `false` to trust, as opposed to just getting rid of it as an arg entirely?

deitch commented on src/cmd/linuxkit/cache/push.go:
```diff
@@ -12,8 +12,6 @@ import (
 // PushWithManifest push an image along with, optionally, a multi-arch index.
 func PushWithManifest(dir string, name, suffix string, pushImage, pushManifest, trust, sign bool) error {
```
We probably should remove `trust` as an arg here.

deitch commented on src/cmd/linuxkit/pkglib/docker.go:
```diff
@@ -20,7 +20,7 @@ import (
 
 const (
-	dctEnableEnv        = "DOCKER_CONTENT_TRUST=1"
+	dctEnableEnv        = ""
```
Why not just get rid of it?

deitch reviewed (changes requested):
Two requests:

1. One of the comment changes is confusing
2. This PR removes the `trust` env var or arg from lots of places, but leaves it in a few places, always setting it to `false` or `""` and not passing it anywhere. We should just complete the process and get rid of those.

dave-tucker commented:
@deitch good catched. most of it was some clumsy `sed` on my behalf :blush: I've gone through, fixed up the comments in question and removed the `trust bool` from `pkg/moby`.

deitch reviewed (dismissed):
LGTM, and thank you for taking care of this.

We should wait for a LGTM from all of the core maintainers.

justincormack commented on docs/packages.md:
```diff
@@ -56,7 +54,7 @@ A package source consists of a directory containing at least two files:
 - If you are on a Mac you also need `docker-credential-osxkeychain.bin`, which comes with Docker for Mac.
-- `make`, `notary`, `base64`, `jq`, and `expect`
+- `make`, `base64`, `jq`, and `expect`
 - A *recent* version of `manifest-tool` which you can build with `make
```
Do we still need this?

deitch replied:
No, we do not. manifest-tool now is embedded as a library inside the linuxkit binary. We no longer exec out to it.

For that matter, what to we use expect, base64 and jq for?

rn replied:
kernel build still uses this

justincormack commented on scripts/push-manifest.sh:
```diff
@@ -62,24 +56,4 @@ OUT=$(manifest-tool $MT_ARGS push from-args \
-    exit 0
-fi
-
-# Extract sha256 and length from the manifest-tool output
```
I think you can remove all the manifest-tool parts above as it is only used here

rn replied:
we can probably remove this script. it;'s only used for kernel build and the alpine base. if we can replace the manifest tool with docker than there is no need for this script

dave-tucker replied:
I'll keep it here. We can remove it in a follow-up PR that removes manifest-tool and related bits

justincormack commented on src/cmd/linuxkit/pkglib/build.go:
```diff
@@ -11,7 +11,7 @@ import (
 
 	"github.com/containerd/containerd/reference"
-	"github.com/google/go-containerregistry/pkg/v1"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
```
thats a horrible name to give it!

justincormack commented on tools/alpine/push-manifest.sh:
```diff
@@ -4,14 +4,11 @@ set -e
 # This script creates a multiarch manifest for the 'linuxkit/alpine'
 # image, pushes and signs it. The manifest is pushed with the tag of
```
remove "signs it"

dave-tucker replied:
Done

justincormack commented on tools/alpine/push-manifest.sh:
```diff
@@ -4,14 +4,11 @@ set -e
 #
-# For signing, DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE must be set.
-#
 # This should all be replaced with 'docker manifest' once it lands.
```
docker manifest did land a long time ago, can we use instead of manifest tool?

deitch replied:
We should consider it as a replacement for the embedded library stuff (although I usually prefer embedded), but that definitely should be subject of a separate PR

rn commented on docs/packages.md:
````diff
@@ -179,56 +173,12 @@ To perform _all_ of the steps at once - build and push out the image for whateve
-notary -s https://notary.docker.io -d ~/.docker/trust addhash -p $IMAGE $TAG $SIZE --sha256 $HASH  -r targets/releases
-```
 
 ##### Getting Size and Hash
````
IIRC the size was only needed for signing. so can probably remove this section as well

dave-tucker replied:
Done

rn commented on kernel/Makefile:
```diff
@@ -143,12 +137,6 @@ push: push_$(2)$(3)$(4)
 forcepush: forcepush_$(2)$(3)$(4)
 show-tags: show-tag_$(2)$(3)$(4)
 
-# 'docker build' with the FROM image supplied as --build-arg
```
I wouldn't delete this comment yet, since it sounds like by removing content trust we can simplify the bits below

dave-tucker replied:
Addressed. Added a FIXME to the top to show that the comment isn't current anymore.

rn reviewed (approved):
Thanks for working on this.
